### ISSUE TITLE
Improvement: Remove requirement for NTP sync'd host

### ISSUE THREAD

**Issue by yuilleb** [storj/core#521]
### Discussion

The NTP client in storj can calculate the difference between network time and host time (delta). I purpose we create a `time` utility class/function to replace instances of Date.now() that is passed to all libraries much like we pass logging utilities.

```
function StorjTime(offset) {
   this.offset  = offset; // set via NTP client delta
}

StorjTime.prototype.setOffset = function(offset) {
  this.offset  = offset; // update the offset any time NTP delta changes
  return this;
};

StorjTime.prototype.getOffset = function(offset) {
  return this.offset; // return the offset in case we need to convert received timestamps to localtime
};

StorjTime.prototype.now = function() {
  return Date.now() + timeOffset;
};
```

Whenever the code needs to get the current unix timestamp, rather than calling `Date.now()` it could call `this._time.now()` which would return our best approximation of an NTP sync'd timestamp.

My hope is this would solve the problems where the average user running windows has a system clock not correctly sync'd with NTP.

What do you guys think?

**Comment by littleskunk** [comment]
kad-quasar is now hop based and don't need a timestamp anymore. core still has a few timestamps like this one: https://github.com/Storj/core/blob/ed734e2a56e84b8d9af6961b7ed42f70e7e6d75c/lib/network/index.js#L521-L523

**Comment by yuilleb** [comment]
@littleskunk I think the Quasar paper still wants the Update messages to be time based ttl's. It should just be the publish messages that are hop based. So keep that in mind that we do sadly need sync'd clocks.

**Comment by MeijeSibbel** [comment]
@bookchin , @super3 , this is starting to become a serious issue, basically +95% of all new users that come to us for help on rocketchat have the issue that their clock is not synchronized correctly. Every single one of them running windows has their clock off by at least 1-10 seconds.

**Comment by littleskunk** [comment]
I don't agree on that. A few user found the reason for this here: https://github.com/Storj/storjshare-daemon/issues/72#issuecomment-279860688

It looks more like we can't run UPnP and NTP at the same time. We have to fix that or this improvement will go on with wrong time sync values.

**Comment by MeijeSibbel** [comment]
@littleskunk , That is even worse, now we can't tell at all from the logs if the NTP sync value is within acceptable range for those running UPnP.

**Comment by littleskunk** [comment]
@MeijeSibbel Will be fixed with #657

**Comment by RichardLitt** [comment]
👋 Hey! Thanks for this contribution. Apologies for the delay in responding!

We've decided to rearchitect Storj, so that we can scale better. You can read more about this decision [here](https://storj.io/blog/2018/01/getting-from-petabytes-to-exabytes-the-road-ahead/). This means that we are entirely focused on v3 at the moment, in the [storj/storj](https://github.com/storj/storj) repository. Our white paper for v3 is coming **very, very** soon - follow along on [the blog](https://storj.io/blog) and in our [Rocketchat](https://community.storj.io/home).

As this repository is part of the v2 network, we're no longer maintaining this repository. I am going to close this for now. If you have any questions, I encourage you to jump on Rocketchat and ask them there. Thanks!
